lib/components/Snackbar/__tests__/snackbar.test.tsx

```diff
@@ -0,0 +1,109 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Snackbar } from "..";
+
+describe("Snackbar", () => {
+    it("renders with default props", () => {
+        const { container } = render(
+            <Snackbar
+                message="Test message"
+                isOpen={true}
+                onClose={jest.fn()}
+            />,
+        );
+        expect(screen.getByText("Test message")).toBeInTheDocument();
+        expect(container).toMatchSnapshot();
+    });
+    it("calls onActionClick when action button is clicked", () => {
+        const onActionClickMock = jest.fn();
+        const { container } = render(
+            <Snackbar
+                message="Test message"
+                actionText="Action"
+                isOpen={true}
+                onActionClick={onActionClickMock}
+                onClose={jest.fn()}
+            />,
+        );
+
+        fireEvent.click(screen.getByText("Action"));
+        expect(onActionClickMock).toHaveBeenCalled();
+        expect(container).toMatchSnapshot();
+    });
+    it("calls onClose when close button is clicked", () => {
+        const onCloseMock = jest.fn();
+        const { container } = render(
+            <Snackbar
+                message="Test message"
+                isOpen={true}
+                onClose={onCloseMock}
+            />,
+        );
+
+        fireEvent.click(screen.getByText("x"));
+        expect(onCloseMock).toHaveBeenCalled();
+        expect(container).toMatchSnapshot();
+    });
+    it("calls onClose after a certain duration when Snackbar is open", async () => {
+        jest.useFakeTimers();
+
+        const onCloseMock = jest.fn();
+        const { container } = render(
+            <Snackbar
+                message="Test message"
+                isOpen={true}
+                onClose={onCloseMock}
+            />,
+        );
+
+        jest.advanceTimersByTime(3000);
+        expect(onCloseMock).toHaveBeenCalled();
+        expect(container).toMatchSnapshot();
+
+        jest.useRealTimers();
```

Comment:
Sorry, do we need real timers here? Maybe `jest.clearAllTimers();`?
Also you add `async`, but there is no `await waitFor`